gen/cheader.tmpl

```diff
@@ -0,0 +1,232 @@
+#else       // defined(WGPU_SHARED_LIBRARY)
+#    define WGPU_EXPORT
+#endif  // defined(WGPU_SHARED_LIBRARY)
+
+#if !defined(WGPU_OBJECT_ATTRIBUTE)
+#define WGPU_OBJECT_ATTRIBUTE
+#endif
+#if !defined(WGPU_ENUM_ATTRIBUTE)
+#define WGPU_ENUM_ATTRIBUTE
+#endif
+#if !defined(WGPU_STRUCTURE_ATTRIBUTE)
+#define WGPU_STRUCTURE_ATTRIBUTE
+#endif
+#if !defined(WGPU_FUNCTION_ATTRIBUTE)
+#define WGPU_FUNCTION_ATTRIBUTE
+#endif
+#if !defined(WGPU_NULLABLE)
+#define WGPU_NULLABLE
+#endif{{"\n" -}}
+
+{{if eq .Name "webgpu"}}
+#include <stdint.h>
+#include <stddef.h>
+{{else}}
+#include "webgpu.h"
+{{end}}
+
+{{- if .Constants}}
+{{-   range .Constants}}
+{{- MComment .Doc 0}}
+#define WGPU_{{.Name | ConstantCase}} ({{.Value | CValue}})
+{{-   end}}
+{{  end}}
+
+{{- if eq .Name "webgpu"}}
+typedef uint32_t WGPUFlags;
+typedef uint32_t WGPUBool;
+{{  end}}
+
+{{- if .Objects}}
+{{-   range .Objects}}
+{{-     if not .IsStruct}}
+typedef struct WGPU{{.Name | PascalCase}}Impl* WGPU{{.Name | PascalCase}} WGPU_OBJECT_ATTRIBUTE;
+{{-     end}}
+{{-   end}}
+{{  end}}
+
+{{- if .Structs}}
+// Structure forward declarations
+{{-   range .Structs}}
+struct WGPU{{.Name | PascalCase}};
+{{-   end}}
+{{  end}}
+
+{{- range $entry := .Enums}}
+{{-   MComment .Doc 0}}
+typedef enum WGPU{{.Name | PascalCase}} {
+{{-   range $entryIndex, $_ := .Entries}}
+{{-     MComment .Doc 4}}
+{{-     $entryValue := 0}}
+{{-     if eq .Value ""}}
+{{-       $entryValue = $entryIndex}}
+{{-     else}}
+{{-       $entryValue = ParseUint .Value 16}}
+{{-     end}}
+    WGPU{{$entry.Name | PascalCase}}_{{.Name | PascalCase}} = {{printf "%s%.4X," $.EnumPrefix $entryValue}}
+{{-   end}}
+    WGPU{{.Name | PascalCase}}_Force32 = 0x7FFFFFFF
+} WGPU{{.Name | PascalCase}} WGPU_ENUM_ATTRIBUTE;
+{{  end}}
+
+{{- range $entry := .Bitflags}}
+{{-   MComment .Doc 0}}
+typedef enum WGPU{{.Name | PascalCase}} {
+{{-   range $entryIndex, $_ := .Entries}}
+{{-     MComment .Doc 4}}
+{{-     $entryValue := ""}}
+{{-     $valueCombination := .ValueCombination}}
+{{-     range $valueIndex, $v := .ValueCombination}}
+{{-       $v = printf "WGPU%s_%s" ($entry.Name | PascalCase) ($v | PascalCase)}}
+{{-       if IsLast $valueIndex $valueCombination}}
+{{-         $entryValue = print $entryValue $v}}
+{{-       else}}
+{{-         $entryValue = print $entryValue $v " | "}}
+{{-       end}}
+{{-     else}}
+{{-       if eq .Value ""}}
+{{-         $entryValue = printf "0x%.8X" (BitFlagValue $entryIndex)}}
+{{-       else}}
+{{-         $entryValue = printf "0x%.8X" (ParseUint .Value 64)}}
+{{-       end}}
+{{-     end}}
+    WGPU{{$entry.Name | PascalCase}}_{{.Name | PascalCase}} = {{$entryValue}},
+{{-   end}}
+    WGPU{{.Name | PascalCase}}_Force32 = 0x7FFFFFFF
+} WGPU{{.Name | PascalCase}} WGPU_ENUM_ATTRIBUTE;
+typedef WGPUFlags WGPU{{.Name | PascalCase}}Flags WGPU_ENUM_ATTRIBUTE;
+{{  end}}
+
+{{- if eq .Name "webgpu"}}
+typedef void (*WGPUProc)(void) WGPU_FUNCTION_ATTRIBUTE;
+{{- end}}
+
```

Comment:
nit: this blank line disappears from the generated result, but I think it would be more readable if it were kept